### PR TITLE
env: upgrade pip on all versions of Python under 3.6.6

### LINKED ISSUE #226: bug: fails on GHA's default pipx
This has suddenly started failing on GitHub Actions, using pipx. It seems to be trying to use `--no-warn-script-location` when such an option does not exist.

```
Run pipx run --spec build pyproject-build
Uninstalling setuptools-39.0.1:
  Successfully uninstalled setuptools-39.0.1
25l25h
Usage:   
  /tmp/build-env-4hxt_nra/bin/python -m pip install [options] <requirement specifier> [package-index-options] ...
  /tmp/build-env-4hxt_nra/bin/python -m pip install [options] -r <requirements file> [package-index-options] ...
  /tmp/build-env-4hxt_nra/bin/python -m pip install [options] [-e] <vcs project url> ...
  /tmp/build-env-4hxt_nra/bin/python -m pip install [options] [-e] <local project path> ...
  /tmp/build-env-4hxt_nra/bin/python -m pip install [options] <archive url/path> ...

no such option: --no-warn-script-location
Traceback (most recent call last):
  File "/opt/pipx/.cache/a354a1c2584f35b/bin/pyproject-build", line 8, in <module>
    sys.exit(entrypoint())
  File "/opt/pipx/.cache/a354a1c2584f35b/lib/python3.6/site-packages/build/__main__.py", line 209, in entrypoint
    main(sys.argv[1:])
  File "/opt/pipx/.cache/a354a1c2584f35b/lib/python3.6/site-packages/build/__main__.py", line 205, in main
    build_package(args.srcdir, outdir, distributions, config_settings, not args.no_isolation, args.skip_dependencies)
  File "/opt/pipx/.cache/a354a1c2584f35b/lib/python3.6/site-packages/build/__main__.py", line 94, in build_package
    _build_in_isolated_env(builder, outdir, distributions, config_settings)
  File "/opt/pipx/.cache/a354a1c2584f35b/lib/python3.6/site-packages/build/__main__.py", line 56, in _build_in_isolated_env
    env.install(builder.build_dependencies)
  File "/opt/pipx/.cache/a354a1c2584f35b/lib/python3.6/site-packages/build/env.py", line 142, in install
    subprocess.check_call(cmd)
  File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/tmp/build-env-4hxt_nra/bin/python', '-Im', 'pip', 'install', '--no-warn-script-location', '-r', '/tmp/build-reqs-paf79l14.txt']' returned non-zero exit status 2.
```

This will start breaking about a dozen libraries for me. :'( And anyone following the Scikit-HEP guides: https://scikit-hep.org/developer/gha_pure#distribution-pure-python-wheels

### PR DESCRIPTION
**Pull request by layday**
Fixes #226.

### CODE REVIEW

**Review by henryiii — APPROVED** [review]
This looks reasonable to me, at least. :)

**Review by gaborbernat — CHANGES_REQUESTED** [review]
A better solution would be to instead set PIP_NO_WARN_SCRIP_LOCATION=1 env-var. This would degrade better to versions without the flag and would not need this update dance.

**Review by gaborbernat — APPROVED** [review]

**Comment by henryiii** [comment]
I think it still should do the update dance, though? As it does need 10+ anyway, correct? This was just a side effect (an avoidable one, as you point out), but it should be using 10+?

**Comment by layday** [comment]
I don't know what we need 10+ for, was it just `--no-warn-script-location`?

**Comment by henryiii** [comment]
I was just basing my observation on the comment in the code about 10 being needed - this used to work just fine before, so maybe swapping for the environment variable and dropping the pip upgrade is better?

I've fixed my immediate issue by swapping ubuntu-latest for ubuntu-20.04, but obviously that's just a workaround.

**Comment by layday** [comment]
Was hoping @gaborbernat might know :)

**Comment by gaborbernat** [comment]
I guess upgrading pip is safer. For example, old pip might not understand correctly the metadata specification of wheels/sdist, which might cause certain dependencies to fail.

**Review by FFY00 — CHANGES_REQUESTED** [review]
Can you add an entry to the changelog? You'll need to re-add the `Unreleased` section, which was removed when I tagged the latest release.
https://github.com/pypa/build/commit/a5a0b3860cff68ac9aad896fad22febd2b8d3718

File: CHANGELOG.rst [hunk 1]
```diff
@@ -3,6 +3,11 @@ Changelog
 +++++++++
 
 
+Unreleased
+==========
+
+- Fix error from unrecognised pip flag on Python 3.6.0 to 3.6.5 (`PR #227`_, Fixes `#226`_)
```
Comment (FFY00) [inline CHANGELOG.rst:9]: You need to add the links as below.

**Review by FFY00 — APPROVED** [review]

**Comment by henryiii** [comment]
Any chance we might get a release? Dependabot is going through, updating all my repos to PyPI publish 1.4.2, and they are breaking due to the fact that build is broken, for example https://github.com/scikit-hep/uhi/pull/5. I can manually change to using ubuntu-20.04, but it would be nice not to have to change a few dozen repos and add a note to our developer docs. Thanks! :)

**Comment by FFY00** [comment]
I will do a bugfix release.

**Comment by henryiii** [comment]
I really appreciate it, thank you very much. :)

**Pull request merged**